v2/protocol/http/protocol.go

```diff
@@ -267,6 +268,7 @@ func (p *Protocol) Receive(ctx context.Context) (binding.Message, error) {
 // Returns non-nil error if the incoming HTTP request fails to parse as a CloudEvent
 // Returns io.EOF if the receiver is closed.
 func (p *Protocol) Respond(ctx context.Context) (binding.Message, protocol.ResponseFn, error) {
+	const connectionReset = "connection reset by peer"
```

Comment:
You can use a typed error here e.g., using `errors.Is()` based on `ECONNRESET` from https://pkg.go.dev/syscall#pkg-constants